packages/shorebird_cli/test/src/artifact_builder_test.dart

```diff
@@ -78,17 +78,80 @@ void main() {
           .thenReturn('/path/to/flutter');
       when(() => shorebirdEnv.flutterRevision).thenReturn('1234');
       when(shorebirdEnv.getShorebirdProjectRoot).thenReturn(Directory(''));
-      when(
-        () => shorebirdAndroidArtifacts.findAab(
-          project: any(named: 'project'),
-          flavor: any(named: 'flavor'),
-        ),
-      ).thenReturn(File('app-release.aab'));
 
       builder = ArtifactBuilder();
     });
 
+    Future<void> verifyCorrectFlutterPubGet(
+      Future<void> Function() testCall,
+    ) async {
+      group('when flutter is installed', () {
```

Comment:
Seems strange to have groups and tests nested within a top-level function list this. Can we annotate this as `@isTest` instead and implement the standard test API?

Reply:
When you say "the standard test API", what do you mean exactly? It's not really a test, but a helper function that invokes runs test to verify shared functionality (that `flutter pub get` does or does not get called after a build method)

Reply:
I meant annotating it with `@isTest` and supplying a description, callback etc.

Reply:
Resolving this based on our discussions 👍

Reply:
Discussed offline, decided this is good as is